common/src/chain/in_mem_block_tree/tree_wrappers.rs

```diff
@@ -0,0 +1,413 @@
+// Copyright (c) 2021-2024 RBB S.r.l
```

Comment:
This module needs its own tests (in particular for the `ensure_node_in_subtree` calls).